types/joi/index.d.ts

```diff
@@ -185,7 +186,7 @@ export interface ValidationErrorFunction {
     (errors: ValidationErrorItem[]): string | ValidationErrorItem | ValidationErrorItem[] | Error;
 }
 
-export interface ValidationResult<T> {
+export interface ValidationResult<T> extends Pick<Promise<T>, 'then' | 'catch'> {
```

Comment:
Maybe just `extends Promise<T>` is okay?

Reply:
I think `extends Pick<Promise<T>, 'then' | 'catch'>` is good enough and exactly the same as the implementation. https://github.com/hapijs/joi/blob/master/lib/types/any/index.js#L660